lti_consumer/plugin/views.py

```diff
@@ -130,3 +144,79 @@ def get_queryset(self):
     def perform_create(self, serializer):
         lti_configuration = self.request.lti_configuration
         serializer.save(lti_configuration=lti_configuration)
+
+    @action(
+        detail=True,
+        methods=['GET'],
+        renderer_classes=[LineItemResultsRenderer]
+    )
+    def results(self, request, *args, **kwargs):
+        """
+        Return a Result list for an LtiAgsLineItem
+
+        Query Parameters:
+          * user_id (string): String external user id representation. Only
+                one record per unique `user_id` will be returned
+          * limit (integer): The maximum number of records to return. Records are
+                sorted with most recent timestamp first
+
+        Returns:
+          * An array of Result records, formatted by LtiAgsResultSerializer
+                and returned with the media-type for LineItemResultsRenderer
+        """
+        line_item = self.get_object()
+        scores = line_item.scores.all().order_by('-timestamp')
+
+        if request.query_params.get('user_id'):
+            scores = scores.filter(user_id=request.query_params.get('user_id'))
+
+        # Only get one record per user
+        # Would use scores.distinct('user_id') here, but MySQL does
+        # not support it distinct with fields
+        results = {}
+        for score in scores:
+            if score.user_id not in results:
+                results[score.user_id] = score
+        results = results.values()
+
+        if request.query_params.get('limit'):
+            results = results[:int(request.query_params.get('limit'))]
+
+        serializer = LtiAgsResultSerializer(
+            list(results),
+            context={'request': self.request},
+            many=True,
+        )
+
+        return Response(serializer.data)
+
+    @action(
+        detail=True,
+        methods=['POST'],
+        parser_classes=[LineItemScoreParser],
+        renderer_classes=[LineItemScoreRenderer]
+    )
+    def scores(self, request, *args, **kwargs):
+        """
+        Return a Result list for an LtiAgsLineItem
+
+        Data:
+          * A JSON object capable of being serialized by LtiAgsScoreSerializer
+
+        Returns:
+          * An copy of the saved record, formatted by LtiAgsScoreSerializer
+                and returned with the media-type for LineItemScoreRenderer
+        """
+        line_item = self.get_object()
+        serializer = LtiAgsScoreSerializer(
+            data=request.data,
+            context={'request': self.request},
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save(line_item=line_item)
```

Comment:
To avoid needing to handle multiple `LtiAgsScore` entries for each user, we can use a single item and modify when new data gets pushed through the score service. 

LTI AGS doesn't require the platform to keep a history of changes to the students score, and just needs to make sure that scores are not updated out of order (looking at the `timestamp` field):
>The platform MUST NOT update a result if the last timestamp on record is later than the incoming score update. It may just ignore the incoming score update, or log it if it maintains any kind of history or for traceability.

That will simplify the implementation in a sense that we won't need to implement signals and clear up old values when a empty score or comment value is sent through the tool.

We can use a `unique_together` relation between the User and the LineItem.